api/templates/query_builder.html

```diff
@@ -427,17 +427,22 @@ <h3>Welcome to the QueryBuilder Page</h3>
             this.finalUrl =
               this.url +
               this.dropdownSelectedAPI +
-              "?" +
-              selectedCountries +
-              "&" +
-              selectedYears +
-              "&Pivot=" +
-              this.dropdownSelectedPivot +
-              "&Theme=" +
-              this.dropdownSelectedTheme +
-              "&Plot=" +
-              this.dropdownSelectedPlot +
-              "&Number=" +
+              "?";
+              if (this.dropdownSelectedAPI != 'stats')
+              this.finalUrl += selectedCountries;
+              this.finalUrl += "&" +
+              selectedYears;
+              if (this.dropdownSelectedAPI == 'json' || this.dropdownSelectedAPI == 'table')
+              this.finalUrl += "&Pivot=" +
+              this.dropdownSelectedPivot;
+              if (this.dropdownSelectedAPI != 'json' && this.dropdownSelectedAPI != 'table')
+              this.finalUrl += "&Theme="+
+              this.dropdownSelectedTheme;
+              if (this.dropdownSelectedAPI == 'graph')
+              this.finalUrl += "&Plot=" +
+              this.dropdownSelectedPlot;
+              if (this.dropdownSelectedAPI =='stats')
+              this.finalUrl += "&Number=" +
```

Comment:
Have you tried building two different queries without reloading?

This won't work then (as far as I recall), because then it will append the same query params two times

Reply:
But why will it ?

Reply:
Yes, this happened, without refreshing, it was taking the previous query params but i thought it was a problem with loading.